src/dlsnode/util/aio/AsyncIO.d

```diff
@@ -144,22 +147,24 @@ class AsyncIO
 
     **************************************************************************/
 
-    public size_t pread (void[] buf, int fd, size_t offset,
+    public size_t pread (void[]* buf, int fd, size_t offset,
```

Comment:
Changing the type of `buf` should not be necessary, you can just use `&buf`.

Reply:
Do you mean that `void[] buf` accepts both `void[]` and `void[]*`?

Reply:
I mean:
```d
public size_t pread (void[] buf, Etc etc)
{
    // ...
    job.user_buffer = &buf;
    // ...
}
```
That way should work, too, and the method signature stays the same so callers don't need to be adapted.

Reply:
Never mind, I understand the pointer can be used after the method has returned.

Reply:
Looking back, there was no need to capture the slice by pointer anyway, since we wouldn't resize original slice anyway. Updated.